src/engine/QueryExecutionTree.cpp

```diff
@@ -104,6 +104,26 @@ size_t QueryExecutionTree::getSizeEstimate() {
   return sizeEstimate_.value();
 }
 
+//_____________________________________________________________________________
+std::optional<std::shared_ptr<QueryExecutionTree>>
+QueryExecutionTree::setPrefilterExprGetUpdatedQetPtr(
+    std::vector<Operation::PrefilterVariablePair> prefilterPairs) const {
+  AD_CONTRACT_CHECK(rootOperation_);
+  std::vector<Operation::PrefilterVariablePair> relevantPairs;
+  relevantPairs.reserve(prefilterPairs.size());
+  VariableToColumnMap varToColMap = getVariableColumns();
+  // Add all the PrefilterVariable values whose Variable value is
+  // contained in the VariableToColumnMap. This is done to avoid that certain
+  // subqueries filter out too much.
+  for (auto& prefilterPair : prefilterPairs) {
+    if (varToColMap.find(prefilterPair.second) != varToColMap.end()) {
```

Comment:
As you get the input vector by value, you can simply mutate it in place by using
`std::erase_if(prefilterPairs, [](const auto& pair){...})`
This saves quite some lines.